devito/mpi/halo_scheme.py

```diff
@@ -481,3 +538,34 @@ def process_loc_indices(raw_loc_indices, directions):
     loc_dirs = {d: v for d, v in directions.items() if d in known}
 
     return frozendict(loc_indices), frozendict(loc_dirs)
+
+
+class HaloTouch(sympy.Function, Reconstructable):
+
+    """
+    A SymPy object representing halo accesses through a HaloScheme.
+    """
+
+    __rargs__ = ('args',)
+    __rkwargs__ = ('halo_scheme',)
+
+    def __new__(cls, *args, halo_scheme=None, **kwargs):
+        obj = super().__new__(cls, *args)
+        obj.halo_scheme = halo_scheme
+        return obj
+
+    def __repr__(self):
+        return "HaloTouch(%s)" % ",".join(f.name for f in self.halo_scheme.fmapper)
+
+    __str__ = __repr__
+
+    def _sympystr(self, printer):
+        return str(self)
```

Comment:
if you have a printer shouldn't it be `printer.doprint(self)` ?

Reply:
I don't remember exactly, but probably not, pretty sure I tried it but didn't eventually obtained what I wanted. Maybe because the printer lacks the handle for `self`, that's why we end up here? Anyway, this is just for the `repr` within larger expressions